src/sizeMapping.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @module sizeMapping
+ */
+import * as utils from './utils';
+
+exports.mapSizes = function(adUnit) {
+  if(!isSizeMappingValid(adUnit.sizeMapping)){
+    return adUnit.sizes;
+  }
+  const width = this.getScreenWidth();
+  if(!width) {
+    //size not detected - get largest value set for desktop
+    const mapping = adUnit.sizeMapping.reduce((prev, curr) => {
+      return prev.minWidth < curr.minWidth ? curr : prev;
+    });
+    if(mapping.sizes) {
+      return mapping.sizes;
+    }
+    return adUnit.sizes;
+  }
+  const sizes = adUnit.sizeMapping.find(sizeMapping =>{
+    return width > sizeMapping.minWidth;
+  }).sizes;
+  utils.logMessage(`AdUnit : ${adUnit.code} resized based on device width to : ${adUnit.sizes}`);
+  return sizes;
+
+};
+
+function isSizeMappingValid(sizeMapping) {
+  if(utils.isArray(sizeMapping) && sizeMapping.length > 0){
+    return true;
+  }
+  utils.logError('sizeMapping needs at least one screen size defined');
```

Comment:
On pages that haven't set up any `sizeMapping`s but are still using `sizes` array(s), this message still gets logged